pysat/tests/classes/cls_instrument_library.py

```diff
@@ -0,0 +1,436 @@
+        instruments = generate_instrument_list(inst_loc=inst_loc,
+                                               user_info=user_info)
+
+        # Find all methods in the standard test class.
+        method_list = [func for func in dir(self)
+                       if callable(getattr(self, func))]
+
+        # Search tests for iteration via pytestmark, update w/ instrument list.
+        for method in method_list:
+            if hasattr(getattr(self, method), 'pytestmark'):
+                # Get list of names of pytestmarks.
+                n_args = len(getattr(self, method).pytestmark)
+                mark_names = [getattr(self, method).pytestmark[j].name
+                              for j in range(0, n_args)]
+                # Add instruments from your library.
+                if 'all_inst' in mark_names:
+                    mark = pytest.mark.parametrize("inst_name",
+                                                   instruments['names'])
+                    getattr(self, method).pytestmark.append(mark)
+                elif 'load_options' in mark_names:
+                    # Prioritize load_options mark if present
+                    mark = pytest.mark.parametrize("inst_dict",
+                                                   instruments['load_options'])
+                    getattr(self, method).pytestmark.append(mark)
+                elif 'download' in mark_names:
+                    mark = pytest.mark.parametrize("inst_dict",
+                                                   instruments['download'])
+                    getattr(self, method).pytestmark.append(mark)
+                elif 'no_download' in mark_names:
+                    mark = pytest.mark.parametrize("inst_dict",
+                                                   instruments['no_download'])
+                    getattr(self, method).pytestmark.append(mark)
+
+        return instruments
+
+    def assert_hasattr(self, obj, attr_name):
+        """Provide useful info if object is missing a required attribute.
+
+        Parameters
+        ----------
+        obj : object
+            Name of object to check.
+        attr_name : str
+            Name of required attribute that must be present in `obj`.
+
+        """
+
+        estr = "Object {:} missing attribute {:}".format(obj.__repr__(),
+                                                         attr_name)
+        assert hasattr(obj, attr_name), estr
+        return
+
+    def assert_isinstance(self, obj, obj_type):
+        """Provide useful info if object is the wrong type.
+
+        Parameters
+        ----------
+        obj : object
+            Name of object to check.
+        obj_type : str
+            Required type of object.
+
+        """
+
+        estr = "Object {:} is type {:}, but should be type {:}".format(
+            obj.__repr__(), type(obj), obj_type)
+        assert isinstance(obj, obj_type), estr
+        return
```

Comment:
These look like they might belong in `utils.testing`.  They also both need a Raises section in the docstring if that is done.